src/main/java/com/zendesk/maxwell/MaxwellConfig.java

```diff
@@ -263,6 +273,30 @@ private void setup(OptionSet options, Properties properties) {
 			}
 		}
 
+		if(this.kafkaPartitionKey != "database") {
+			LOGGER.warn("kafka_partition_by is deprecated, please use producer_partition_by");
+			this.producerPartitionKey = this.kafkaPartitionKey;
+		} else {
+			this.producerPartitionKey = fetchOption("producer_partition_by", options, properties, "database");
```

Comment:
I don't particularly care if the `kafka_` set of options take precedence or the `producer_` ones do, but this might read cleaner if one just clobbered the other, ie:

```
  this.producerPartitionKey = fetchOption("producer_partition_by", options, properties, "database");
  this.producerPartitionColumns = fetchOption("producer_partition_columns", options, properties, null);
  ...
  if(!this.kafkaPartitionKey.equals("database")) {
    LOGGER()
  }
  if(this.kafka...)
```